engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/testmode/TestDataPreparer.scala

```diff
@@ -0,0 +1,46 @@
+package pl.touk.nussknacker.engine.testmode
+
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.deployment.TestProcess.TestData
+import pl.touk.nussknacker.engine.api.process.{RunMode, SourceTestSupport}
+import pl.touk.nussknacker.engine.compile.ExpressionCompiler
+import pl.touk.nussknacker.engine.compile.nodecompilation.NodeCompiler
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.graph.node.SourceNodeData
+import pl.touk.nussknacker.engine.resultcollector.PreventInvocationCollector
+import pl.touk.nussknacker.engine.spel.SpelExpressionParser
+
+case class ParsedTestData(samples: List[AnyRef])
+
+class TestDataPreparer(modelData: ModelData) {
+
+  private val nodeCompiler = {
+    val expressionCompiler = ExpressionCompiler.withoutOptimization(modelData).withExpressionParsers {
+      case spel: SpelExpressionParser => spel.typingDictLabels
+    }
+    new NodeCompiler(modelData.processWithObjectsDefinition,
+      expressionCompiler, modelData.modelClassLoader.classLoader, PreventInvocationCollector, RunMode.Normal)
+  }
+
+  def prepareDataForTest(espProcess: EspProcess, testData: TestData): ParsedTestData = {
+    val testParserForSource = (espProcess.roots.map(_.data).collect {
+      case e: SourceNodeData => e
```

Comment:
I thought it can be done in single pattern match - without collect - but it cant ;)